flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/reconciler/sessionjob/SessionJobReconciler.java

```diff
@@ -127,14 +128,14 @@ public DeleteControl cleanupInternal(FlinkResourceContext<FlinkSessionJob> ctx)
                                     : UpgradeMode.STATELESS;
                     cancelJob(ctx, upgradeMode);
                 } catch (ExecutionException e) {
-                    final var cause = e.getCause();
-
-                    if (cause instanceof FlinkJobNotFoundException) {
+                    if (ExceptionUtils.indexOfType(e, FlinkJobNotFoundException.class) != -1) {
```

Comment:
IIUC, we wanna find is there `FlinkJobNotFoundException` inside of e, right?

If so,  `org.apache.flink.util.ExceptionUtils#findThrowable(e, FlinkJobNotFoundException.class).isPresent()` may be better here. It's easy to understand to check the index.

WDYT?

Reply:
Also, would you mind adding a unit test to check whether it is expected when the sub cause is `FlinkJobNotFoundException`?

Reply:
> Also, would you mind adding a unit test to check whether it is expected when the sub cause is

We don't need to add an e2e or real session job UT, only need to check the exception can be caught, right?

Reply:
It's up to you. Both of e2e ITcase or Unit test are fine for me.

Reply:
I changed the mock service to follow the real behavior.